client/my-sites/plugins/plugins-list/index.jsx

```diff
@@ -0,0 +1,402 @@
+			this.refs.dialog.open();
+		}
+	},
+
+	// Renders
+	render() {
+		const itemListClasses = classNames( 'list-cards-compact', 'plugins-list', {
+			'is-bulk-editing': this.state.bulkManagement
+		} );
+
+		if ( this.props.isPlaceholder ) {
+			return (
+				<div className="plugins-list">
+					<SectionHeader key="plugins-list__section-placeholder" label={ this.props.header } className="plugins-list__section-actions is-placeholder" />
+					<div className={ itemListClasses }>{ this.renderPlaceholders() }</div>
+				</div>
+				)
+		}
+
+		if ( isEmpty( this.props.plugins ) ) {
+			return null;
+		}
+
+		return (
+			<div className="plugins-list" >
+				<PluginsListHeader label={ this.props.header }
+					isWpCom={ this.props.isWpCom }
+					isBulkManagementActive={ !! this.state.bulkManagement }
+					sites={ this.props.sites }
+					plugins={ this.props.plugins }
+					selected={ this.getSelected() }
+					toggleBulkManagement={ this.toggleBulkManagement }
+					updateAllPlugins={ this.updateAllPlugins }
+					pluginUpdateCount={ this.getPluginUpdateCount() }
+					activateSelected={ this.activateSelected }
+					deactiveAndDisconnectSelected={ this.deactiveAndDisconnectSelected }
+					deactivateSelected={ this.deactivateSelected }
+					setAutoupdateSelected={ this.setAutoupdateSelected }
+					unsetAutoupdateSelected={ this.unsetAutoupdateSelected }
+					removePluginNotice={ this.removePluginNotice }
+					setSelectionState={ this.setBulkSelectionState }
+					haveActiveSelected={ this.props.plugins.some( this.filterSelection.active.bind( this ) ) }
+					haveInactiveSelected={ this.props.plugins.some( this.filterSelection.inactive.bind( this ) ) } />
+				<div className={ itemListClasses }>{ this.props.plugins.map( this.renderPlugin ) }</div>
+				{ ! this.props.isWpCom && <DisconnectJetpackDialog ref="dialog" site={ this.props.site } sites={ this.props.sites } redirect="/plugins" /> }
+			</div>
+		);
+	},
+
+	renderPlugin( plugin ) {
+		const selectThisPlugin = this.togglePlugin.bind( this, plugin );
+		return (
+			<PluginItem
+				key={ plugin.slug }
+				hasAllNoManageSites={ this.hasNoSitesThatCanManage( plugin ) }
+				plugin={ plugin }
+				sites={ plugin.sites }
+				progress={ this.state.notices.inProgress.filter( log => log.plugin.slug === plugin.slug ) }
+				errors={ this.state.notices.errors.filter( log => log.plugin && log.plugin.slug === plugin.slug ) }
+				notices={ this.state.notices }
+				isSelected={ this.isSelected( plugin ) }
+				isSelectable={ this.state.bulkManagement }
+				onClick={ selectThisPlugin }
+				selectedSite={ this.props.selectedSite }
+				pluginLink={ '/plugins/' + encodeURIComponent( plugin.slug ) + ( plugin.wpcom ? '/business' : '' ) + this.siteSuffix() } />
+		);
+	},
+
+	renderPlaceholders() {
+		const placeholderCount = this.props.isWpCom ? 3 : 16;
+		return [ ... Array( placeholderCount ).keys() ].map( i => <PluginItem key={ 'placeholder-' + i } /> );
```

Comment:
it looks like we just need a range of numbers here, which **lodash** can provide for us clearly - it took me a minute to figure out why we were creating an array from a scalar number and then getting its keys

```js
import range from 'lodash/utility/range';

return range( placeHolderCount ).map( … );
```